tests/cxx/quoted_headers_first.cc

```diff
@@ -0,0 +1,36 @@
+//===--- prefix_header_operator_new.cc - test input file for iwyu ---------===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+
+// Tests that IWYU will respect the --quoted_headers_first option.
+
+#include <iostream>
+#include <list>
+#include <map>
+#include <memory>
+#include "subfolder/indirect_subfolder.h"
```

Comment:
Hmm, I wonder if it would be beneficial to test this with an associated header, too? To see that associated headers (usually included with quotes) are sorted before other quoted headers. Also, that would provide an opportunity to make sure `--quoted_headers_first` works in header files too (but then you'd probably need to add another quoted include in this `.cc` file).

Reply:
Also, might be a good idea to try this with a PCH.

Reply:
Done, including both associated header and pch.